gossip/tips_cache.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include "gossip/tips_cache.h"
+#include "utils/handles/rand.h"
+
+/*
+ * Private functions
+ */
+
+static retcode_t tips_cache_fifo_add(hash243_set_t* const set,
+                                     size_t const capacity,
+                                     flex_trit_t const* const tip) {
+  retcode_t ret = RC_OK;
+
+  if (set == NULL || tip == NULL) {
+    return RC_NULL_PARAM;
+  }
+
+  if (hash243_set_size(set) >= capacity) {
+    if ((ret = hash243_set_remove_entry(set, *set)) != RC_OK) {
+      return ret;
+    }
+  }
+
+  return hash243_set_add(set, tip);
+}
+
+static retcode_t tips_cache_random_tip_from_set(hash243_set_t* const set,
```

Comment:
Move this outside? This is rather a generic `random_hash_from_hash243_set` method?